### DISCUSSION TITLE
Courses not showing up in PrairieLearn production

### DISCUSSION THREAD

**Discussion by firasm** [PrairieLearn/PrairieLearn#6042] [Q&A] [answered]
Hello, 

I'm trying to get a production PL instance set up, and I'm making some progress (thanks to help from @nwalters512 and [@Prostigus](https://github.com/Prostigus) so far!)

We've got Google auth setup with myself as admin, and are running in "Production": 
<img width="846" alt="Screen Shot 2022-07-27 at 8 36 25 PM" src="https://user-images.githubusercontent.com/2507459/181414970-b45b3331-62f7-43bf-b100-f781886381cd.png">

But I'm having trouble having my courses show up.

In my `config.json` file, I have the course repos added:

```
"config.courseDirs": ["pl_courses/pl-ubc-phys112", "pl_courses/pl-ubc-phys122", "exampleCourse"],
"config.coursesRoot" = 'PrairieLearn/pl_courses';
```

There is some confusion on whether it's [`config.courseDirs`](https://github.com/PrairieLearn/PrairieLearn/blob/master/lib/config.js#L38) or just [`courseDirs`](https://github.com/PrairieLearn/PrairieLearn/blob/2f3b359987bebf00d45a8c32d58e3b302d117c4a/docs/configJson.md#setting-course-directories), I think I've tried both and it doesn't seem to work.

I am fairly confident my docker volumes setup correctly, because I've bash'd into it and I can see the course repos in the docker container:

```
bash-4.2# ls PrairieLearn/pl_courses/
pl-ubc-phys112	pl-ubc-phys122
```

and 

```
bash-4.2# ls PrairieLearn
admin_queries  database  exampleCourse .. ... ...
... ... ... 
```

and even the `exampleCourse` doesn't seem to show up.

I wondered if this was permissions related, and I see this:

```
drwxr-xr-x. 6 2602723  887 141 Jun 28 02:50 pl-ubc-phys112
drwxr-xr-x. 6 2602723  887 141 Jun 28 02:50 pl-ubc-phys122
```

Is there a step I'm missing somewhere to get the courses show up?

**Comment by nwalters512** [comment, marked as answer]
To first answer the question about the config: you don't need to prefix the keys with `config.`, e.g. just `courseDirs` is the correct key. The file you linked to from `config.courseDirs` is a JavaScript file, and `config` is the name of the object that holds all the config values.

To answer the main question here: `courseDirs` is only useful in development mode. That is, it's only used in the `instructorLoadFromDisk` page:

https://github.com/PrairieLearn/PrairieLearn/blob/3f0a2de113a6cbe826471a2e84af562ff7d1064b/pages/instructorLoadFromDisk/instructorLoadFromDisk.js#L43

And that page is only made available in development mode:

https://github.com/PrairieLearn/PrairieLearn/blob/3f0a2de113a6cbe826471a2e84af562ff7d1064b/server.js#L538

This is by design: in production environments, we don't maintain a hardcoded list of courses in the config file. Instead, you need to visit the admin page and use the "Add course" button. You can then fill in the appropriate information, including the Git repository where it should be cloned from.

**Reply by firasm** [reply]
Perfect ! Thank you! This definitely explains a lot. I was just expecting to have the git repo in the folder and have it just work 😅.
